UPDATING.md

```diff
@@ -0,0 +1,126 @@
+# Upgrade / Migration Guide
+
+## Version 1.1.5 to 1.2.0
+
+We have made many **breaking changes** in the version 1.2 release of this SDK.
+
+In this guide we aim to highlight the main differences you will encounter when migrating your code from the interfaces we were offering prior to the [version 1.2.0 release](https://github.com/ably/ably-go/releases/tag/v1.2.0).
+
+These include:
+
+- Changes to numerous function and method signatures - these are breaking changes, in that your code will need to change to use them in their new form
+- Adoption of the [Context Concurrency Pattern](https://blog.golang.org/context) using the [context package](https://pkg.go.dev/context) - e.g. [ably.RealtimeChannel.Subscribe](https://pkg.go.dev/github.com/ably/ably-go/ably#RealtimeChannel.Subscribe)
+- Adoption of the [Functional Options Pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) - e.g. [ably.ClientOption](https://pkg.go.dev/github.com/ably/ably-go/ably#ClientOption)
+
+### Asynchronous Operations and the `Context` Concurrency Pattern
+
+We've now moved to a model where methods are blocking in nature, cancellable with a [Context](https://pkg.go.dev/context#Context).
+
+We use `ctx` in examples to refer to a `Context` instance that you create in your code.
+
+For robust, production ready applications you will rarely (actually, probably _never_) want to create your `Context` using the basic [Background](https://pkg.go.dev/context#Background) function, given it cannot be cancelled and remains for the lifecycle of the program. Instead, you should use [WithTimeout](https://pkg.go.dev/context#WithTimeout) or [WithDeadline](https://pkg.go.dev/context#WithDeadline).
+
+For example, a context can be created with a 10 second timeout like this:
+
+```go
+context.WithTimeout(context.Background, 10 * time.Second)
+```
+
+Adding the necessary code to stay on top of cancellation, you will need something like this:
+
+```go
+ctx, cancelFunc := context.WithTimeout(context.Background(), 10 * time.Second)
+defer cancelFunc()
+```
+
+This way, the context can be cancelled at the close of the function.
+
+### Client Options now has a Functional Interface
+
+Previously you _were_ instantiating a client using a `ClientOptions` instance created using the `NewClientOptions` function:
+
+```go
+client, err := ably.NewRealtime(ably.NewClientOptions("xxx:xxx"))
+```
+
+You **now** need to use the new functional interface:
+
+```go
+client, err := ably.NewRealtime(ably.WithKey("xxx:xxx"))
+```
+
+Methods on `ClientOptions`
```

Comment:
This sentence seems incomplete. Perhaps:

For a full list of client options, see [all functions prefixed `With` that return `ClientOption`](https://pkg.go.dev/github.com/ably/ably-go@v1.2.0/ably#ClientOption).

Reply:
Oh, sorry about that, you're absolutely right. I was intending to link out to where the other options could be found, perhaps even going as far as to map spec'd concepts to the Go equivalents. I shall revisit.

Reply:
Fixed: https://github.com/ably/ably-go/pull/364/commits/5c39e03cfd1689f5986081664aeb618479153fa2